Source/DafnyCore/AST/Expressions.cs

```diff
@@ -104,7 +105,10 @@ public abstract class Expression : INode {
         } else {
```

Comment:
I think that's perfectly acceptable to move RangeToken to INode. For the second part, I prefer that RangeToken is computed lazily, because it depends on sub-expression ranges, format tokens (I still need to migrate them to OwnedTokens) and OwnedTokens.